### PR TITLE
Replace node_redis with ioredis

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

- Replaces the `node_redis` library with `ioredis` due to it's clustering and promise support.

### CODE REVIEW

File: services/cache.js [hunk 1]
```diff
@@ -114,93 +59,77 @@ cache.init = async () => {
   // Create the redis instance.
   cache.client = redis.createClient();
 
-  // Load the INCR_SCRIPT and DECR_SCRIPT into Redis.
-  let [incrScriptHash, decrScriptHash] = await Promise.all([
-    loadScript('INCR_SCRIPT', INCR_SCRIPT),
-    loadScript('DECR_SCRIPT', DECR_SCRIPT)
-  ]);
+  // This is designed to increment a key and add an expiry iff the key already
```
Comment (alinhle) [inline services/cache.js:62]: _iff_ I was a boy 🎶

File: services/cache.js [hunk 2]
```diff
@@ -114,93 +59,77 @@ cache.init = async () => {
...
-  cache.client
-    .evalsha(cache.INCR_SCRIPT_HASH, 1, kf(key), expiry, (err) => {
-      if (err) {
-        return reject(err);
-      }
-
-      return resolve();
-    });
-});
+cache.incr = async (key, expiry, kf = keyfunc) => cache.client.increx(kf(key), expiry);
```
Comment (alinhle) [inline services/cache.js:95]: So nice

File: services/redis.js [hunk 3]
```diff
@@ -64,25 +65,15 @@ const connectionOptions = {
 };
 
 const createClient = () => {
-  let client = redis.createClient(connectionOptions);
+  let client = new Redis(connectionOptions);
 
   // Attach the monitors that will print debug messages to the console.
   attachMonitors(client);
```
Comment (alinhle) [inline services/redis.js:71]: 👍

**Comment by alinhle** [comment]
Doing these changes must have felt good 😄

**Review by alinhle — APPROVED** [review]

File: services/cache.js (hunk 1 above)
Comment (wyattjoh) [reply services/cache.js:62]: I use _iff_ as shorthand for _if and only if_ 😄

**Pull request merged**